go/lang/security/audit/unsafe-reflect-by-name.yaml

```diff
@@ -0,0 +1,27 @@
+rules:
+- id: unsafe-reflect-by-name
+  patterns:
+  - pattern-either:
+    - pattern: |
+        $SMTH.MethodByName($NAME,...)
+    - pattern: |
+        $SMTH.FieldByName($NAME,...)
+  - pattern-not: |
+      $SMTH.MethodByName("...",...)
+  - pattern-not: |
+      $SMTH.FieldByName("...",...)
```

Comment:
Would you happen to know if constant propagation will make it so we don't match cases like:
```
constvar1 = "const"
catch = valueJ.FieldByName(constvar1).String()
```

Reply:
yep :+1: constant propagation works really good in this case!